scripts/assemble_figure.py

```
@@ -0,0 +1,376 @@
+        return library_root / value
+
+    json_index = library_root / "library" / "index.json"
+    if not json_index.exists():
+        return library_root / value
+    try:
+        records: list[dict] = json.loads(json_index.read_text())
+        for r in records:
+            if r.get("id") == value:
+                return library_root / "library" / r["file"]
+    except Exception:
+        pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)